horovod/common/parameter_manager.h

```diff
@@ -32,6 +32,9 @@
 namespace horovod {
 namespace common {
 
+class Controller;
```

Comment:
Can we get rid of these?  Let's just include `controller.h` instead.  Also, we should be able to remove the `mpi.h` include.

Reply:
Yes. The dependency of parameter manager on the controller is removed.